lib/logstash/inputs/rabbitmq.rb

```diff
@@ -93,15 +97,38 @@ def declare_queue
       def consume!
         # we manually build a consumer here to be able to keep a reference to it
         # in an @ivar even though we use a blocking version of HB::Queue#subscribe
-        @consumer = @hare_info.queue.build_consumer(:block => true) do |metadata, data|
-          @codec.decode(data) do |event|
-            decorate(event)
-            @output_queue << event if event
+
+        # The logic here around resubscription might seem strange, but its predicated on the fact
+        # that we rely on MarchHare to do the reconnection for us with auto_reconnect.
+        # Unfortunately, while MarchHare does the reconnection work it won't re-subscribe the consumer
```

Comment:
@michaelklishin if you run logstash with this config: https://github.com/logstash-plugins/logstash-input-rabbitmq/blob/master/examples/rabbitmq_stdout.conf and restart rabbitmq while logstash is running logstash will exit because the consumer will stop blocking and exit.